module/VuFindSearch/src/VuFindSearch/Backend/Blender/Response/Json/RecordCollection.php

```diff
@@ -0,0 +1,441 @@
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   David Maus <maus@hab.de>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org
+ */
+namespace VuFindSearch\Backend\Blender\Response\Json;
+
+use VuFindSearch\Backend\EDS\Response\RecordCollection as EDSRecordCollection;
+use VuFindSearch\Backend\Solr\Response\Json\Facets;
+use VuFindSearch\Response\RecordCollectionInterface;
+
+/**
+ * JSON-based record collection for records from multiple sources.
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   David Maus <maus@hab.de>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org
+ */
+class RecordCollection
+    extends \VuFindSearch\Backend\Solr\Response\Json\RecordCollection
+{
+    /**
+     * Blender configuration
+     *
+     * @var \Laminas\Config\Config
+     */
+    protected $config;
+
+    /**
+     * Mappings configuration
+     *
+     * @var array
+     */
+    protected $mappings;
+
+    /**
+     * Backends to be used for initial results
+     *
+     * @var array
+     */
+    protected $initialResults;
+
+    /**
+     * Any errors encountered
+     *
+     * @var array
+     */
+    protected $errors = [];
+
+    /**
+     * Constructor
+     *
+     * @param \Laminas\Config\Config $config   Configuration
+     * @param array                  $mappings Mappings configuration
+     */
+    public function __construct($config = null, $mappings = [])
+    {
+        $this->config = $config;
+        $this->mappings = $mappings;
+        $this->response = static::$template;
+        $this->initialResults = isset($this->config->Blending->initialResults)
+            ? $this->config->Blending->initialResults->toArray()
+            : [];
+    }
+
+    /**
+     * Initialize blended results
+     *
+     * Creates a record list from 0 to $limit
+     *
+     * @param array $collections Array of record collections
+     * @param int   $limit       Result limit
+     * @param int   $blockSize   Blending block size
+     * @param int   $totalCount  Total result count
+     *
+     * @return array Remaining records keyed by backend identifier
+     */
+    public function initBlended(
+        array $collections,
+        int $limit,
+        int $blockSize,
+        int $totalCount
+    ): array {
+        $this->response = static::$template;
+        $this->response['response']['numFound'] = $totalCount;
+        $this->rewind();
+
+        if (!$collections) {
+            return [];
+        }
+
+        $backendRecords = [];
+        foreach ($collections as $backendId => $collection) {
```

Comment:
Would a `collectBackendRecords` support method be appropriate here? Looks like it would factor out pretty cleanly.

Reply:
Added.